core/logger/splunk.go

```diff
@@ -0,0 +1,241 @@
+	ETH_KEY_EXPORTED = "ETH_KEY_EXPORTED"
+	ETH_KEY_DELETED  = "ETH_KEY_DELETED"
+
+	P2P_KEY_CREATED  = "P2P_KEY_CREATED"
+	P2P_KEY_IMPORTED = "P2P_KEY_IMPORTED"
+	P2P_KEY_EXPORTED = "P2P_KEY_EXPORTED"
+	P2P_KEY_DELETED  = "P2P_KEY_DELETED"
+
+	VRF_KEY_CREATED  = "VRF_KEY_CREATED"
+	VRF_KEY_IMPORTED = "VRF_KEY_IMPORTED"
+	VRF_KEY_EXPORTED = "VRF_KEY_EXPORTED"
+	VRF_KEY_DELETED  = "VRF_KEY_DELETED"
+
+	TERRA_KEY_CREATED  = "TERRA_KEY_CREATED"
+	TERRA_KEY_IMPORTED = "TERRA_KEY_IMPORTED"
+	TERRA_KEY_EXPORTED = "TERRA_KEY_EXPORTED"
+	TERRA_KEY_DELETED  = "TERRA_KEY_DELETED"
+
+	SOLANA_KEY_CREATED  = "SOLANA_KEY_CREATED"
+	SOLANA_KEY_IMPORTED = "SOLANA_KEY_IMPORTED"
+	SOLANA_KEY_EXPORTED = "SOLANA_KEY_EXPORTED"
+	SOLANA_KEY_DELETED  = "SOLANA_KEY_DELETED"
+
+	ETH_TRANSACTION_CREATED    = "ETH_TRANSACTION_CREATED"
+	TERRA_TRANSACTION_CREATED  = "TERRA_TRANSACTION_CREATED"
+	SOLANA_TRANSACTION_CREATED = "SOLANA_TRANSACTION_CREATED"
+
+	JOB_CREATED = "JOB_CREATED"
+	JOB_DELETED = "JOB_DELETED"
+
+	CHAIN_ADDED        = "CHAIN_ADDED"
+	CHAIN_SPEC_UPDATED = "CHAIN_SPEC_UPDATED"
+	CHAIN_DELETED      = "CHAIN_DELETED"
+
+	CHAIN_RPC_NODE_ADDED   = "CHAIN_RPC_NODE_ADDED"
+	CHAIN_RPC_NODE_DELETED = "CHAIN_RPC_NODE_DELETED"
+
+	BRIDGE_CREATED = "BRIDGE_CREATED"
+	BRIDGE_UPDATED = "BRIDGE_UPDATED"
+	BRIDGE_DELETED = "BRIDGE_DELETED"
+
+	FORWARDER_CREATED = "FORWARDER_CREATED"
+	FORWARDER_DELETED = "FORWARDER_DELETED"
+
+	EXTERNAL_INITIATOR_CREATED = "EXTERNAL_INITIATOR_CREATED"
+	EXTERNAL_INITIATOR_DELETED = "EXTERNAL_INITIATOR_DELETED"
+
+	JOB_PROPOSAL_SPEC_APPROVED = "JOB_PROPOSAL_SPEC_APPROVED"
+	JOB_PROPOSAL_SPEC_UPDATED  = "JOB_PROPOSAL_SPEC_UPDATED"
+	JOB_PROPOSAL_SPEC_CANCELED = "JOB_PROPOSAL_SPEC_CANCELED"
+	JOB_PROPOSAL_SPEC_REJECTED = "JOB_PROPOSAL_SPEC_REJECTED"
+
+	CONFIG_UPDATED              = "CONFIG_UPDATED"
+	CONFIG_SQL_LOGGING_ENABLED  = "CONFIG_SQL_LOGGING_ENABLED"
+	CONFIG_SQL_LOGGING_DISABLED = "CONFIG_SQL_LOGGING_DISABLED"
+	GLOBAL_LOG_LEVEL_SET        = "GLOBAL_LOG_LEVEL_SET"
+
+	JOB_ERROR_DISMISSED = "JOB_ERROR_DISMISSED"
+	JOB_RUN_SET         = "JOB_RUN_SET"
+
+	ENV_NONCRITICAL_ENV_DUMPED = "ENV_NONCRITICAL_ENV_DUMPED"
+
+	UNAUTHED_RUN_RESUMED = "UNAUTHED_RUN_RESUMED"
+)
+
+type splunkLogger struct {
+	logger      Logger
+	splunkToken string
+	splunkURL   string
+	developFlag bool
+	hostname    string
+	localIP     string
+}
+
+func newSplunkLogger(logger Logger, splunkToken string, splunkURL string, hostname string, chainlinkDev bool) Logger {
+	// Initialize and return Splunk logger struct with required state for HEC calls
+	return &splunkLogger{
+		logger:      logger.Named("Splunk"),
+		splunkToken: splunkToken,
+		splunkURL:   splunkURL,
+		developFlag: chainlinkDev,
+		hostname:    hostname,
+		localIP:     getLocalIP(),
+	}
+}
+
+func (l *splunkLogger) Auditf(eventID string, data map[string]interface{}) {
+	// goroutine to async POST to splunk HTTP Event Collector (HEC)
+	go l.postLogToSplunk(eventID, data)
```

Comment:
Do we need to worry about these goroutines leaking?

Reply:
Hmm, the only line that might not terminate is the client.Do call, but with the defined Timeout [here](https://github.com/smartcontractkit/chainlink/pull/6782/files#diff-c5f6219e3f94e2249b6d2c89b70420e658a260cb01a7cbc6d203f05b6d8c9654R190). This should be good as long as no other logger implementation introducing some hanging behaviour, but we should catch that in a review if it comes up